docs/samples/gcp-iap/README.md

```diff
@@ -0,0 +1,137 @@
+3. You are using a fairly recent version of istio (istio-networking is v0.11.2+ )
+4. You are using a recent version of KFServing (v0.3+)
+
+### Overview
+This example shows how the [sklearn iris sample](https://github.com/kubeflow/kfserving/tree/master/docs/samples/sklearn) can be adapted to support GCP IAP.  We will use a pre-trained model that is hosted on a public gcs bucket.  The model predicts a binary classificaton of iris specie using 4 numerical features.
+
+In this example we will:
+ 1. **Enable GCP IAP**
+ 1. **Change firewall rules (for private k8s clusters)**
+ 1. **Create the inference service**
+ 1. **Test the private predict endpoint (using port-forwarding)**
+ 1. **Expose the inference service externally using an additional Istio Virtual Service**
+ 1. **Test the external predict endpoint (using iap_request.py)**
+
+Step 5. will become unnecessary when this [kfserving issue](https://github.com/kubeflow/kfserving/issues/824) is resolved.
+
+### Enable GCP IAP for k8s
+You have followed the steps to [Set up OAuth for Cloud IAP](https://www.kubeflow.org/docs/gke/deploy/oauth-setup/)
+
+We assume that you have created Kubeflow profiles and that [GCP workload identity](https://www.kubeflow.org/docs/gke/authentication/) is configured.
+
+This may not be essential for this example, but is recommended as it maps GCP service-accounts (understood by the IAP) to kubernetes service-accounts (understood by kubernetes and istio RBAC).
+
+### Change firewall rules (for private k8s clusters) 
+If you are running Kubeflow in a [private GKE cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters) it will be configured with a restrictive firewall.  
+You will need to ensure that the k8s master node(s) can talk to the following k8s worker node(s) ports: 
+ - 6443 (for cert-manager) 
+ - 8443 (for probes such as kfserving)
+
+To create these rules firewall rules, modify and run:
+
+```
+gcloud compute firewall-rules create kubeflow-webhook-probe --allow=tcp:8443 --target-tags kubeflow-worker --direction INGRESS --network default --priority 1000 --source-ranges 172.16.0.0/28
+
+gcloud compute firewall-rules create kubeflow-cert-manager --allow=tcp:8443 --target-tags kubeflow-worker --direction INGRESS --network default --priority 1000 --source-ranges 172.16.0.0/28
+
+```
+Be careful to check whether the **source-ranges**, **target-tags** and **network** are suitable for your environment.  e.g. it assumes your worker nodes have been tagged with `kubeflow-worker` tags.
+
+
+### Create inference service
+
+To deploy the inference service apply the inferenceservice CRD:
+```
+kubectl apply -f sklearn-iap-no-authz.yaml
+```
+
+Expected Output
+```
+$ inferenceservice.serving.kubeflow.org/sklearn-iap created
+```
+
+As well as the InferenceService, the kfserving webhook typically creates a Deployment, Replicaset, Pod(s) + Knative Service, Route, Configuration and Revision!
```

Comment:
might want to rephrase a bit, KFServing controller creates the `Knative Service` and `Knative Service` creates the configuration, revision and route.